operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -193,25 +197,46 @@ protected Parameters doInvoke(FHIROperationContext operationContext, Class<? ext
             MultivaluedMap<String, String> searchParameters = queryParameters;
             if (startOrEndProvided  && !SUPPORT_CLINICAL_DATE_QUERY.contains(compartmentType)) {
                 LOG.finest("The request specified a '" + START_QUERY_PARAMETER + "' and/or '" + END_QUERY_PARAMETER + "' query parameter. They are not valid for resource type '" + compartmentType + "', so will be ignored.");
                 searchParameters = queryParametersWithoutDates;
             }
-
+            
+            // Look up which extra resources should be returned
+            List<String> extraResources = FHIRConfigHelper.getStringListProperty("fhirServer/operations/everything");
+            
             // Need to construct a new version of the map each time through the loop to edit
             MultivaluedMap<String, String> tempSearchParameters = 
                     new MultivaluedHashMap<String, String>(searchParameters);
             
             Bundle results = null;
             int currentResourceCount = 0;
             try {
-                addIncludesSearchParameters(compartmentType, tempSearchParameters);
+                // Don't need to add more search parameters if the config section isn't there or is empty
+                if (extraResources != null && !extraResources.isEmpty()) {
+                    addIncludesSearchParameters(compartmentType, tempSearchParameters, extraResources);
+                }
                 results = resourceHelper.doSearch(compartmentType, PATIENT, logicalId, tempSearchParameters, null, null);
-                currentResourceCount = results.getTotal().getValue();
+                // Only add resources we don't already have and keep track of what we've found so far,
+                // otherwise, we were getting duplicate entries from the _includes
+                int countBeforeAddingNewResources = allEntries.size();
+                for (Entry entry: results.getEntry()) {
+                    String externalIdentifier = ModelSupport.getTypeName(entry.getResource().getClass()) + "/" + entry.getResource().getId();
+                    if (!resourceIds.contains(externalIdentifier)) {
+                        resourceIds.add(externalIdentifier);
+                        allEntries.add(entry);
+                    }
+                }
+                // Don't need to do additional reads if the config section isn't there or is empty
+                if (extraResources != null && !extraResources.isEmpty()) {
+                    readsOfAdditionalAssociatedResources(compartmentType, results.getEntry(), allEntries, resourceIds, searchParameters, resourceHelper, extraResources);
+                }
+                currentResourceCount = allEntries.size() - countBeforeAddingNewResources;
                 totalResourceCount += currentResourceCount;
                 LOG.finest("Got " + compartmentType + " resources " + currentResourceCount + " for a total of " + totalResourceCount);
```

Comment:
Can you wrap this in an isLoggable please. And thank you

Reply:
Done.